hypervisor/src/kvm/mod.rs

```diff
@@ -1174,38 +1176,37 @@ impl hypervisor::Hypervisor for KvmHypervisor {
         HypervisorType::Kvm
     }
 
-    ///
-    /// Create a Vm of a specific type using the underlying hypervisor, passing memory size
-    /// Return a hypervisor-agnostic Vm trait object
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// # use hypervisor::kvm::KvmHypervisor;
-    /// use hypervisor::kvm::KvmVm;
-    /// let hypervisor = KvmHypervisor::new().unwrap();
-    /// let vm = hypervisor.create_vm_with_type_and_memory(0).unwrap();
-    /// ```
-    fn create_vm_with_type_and_memory(
-        &self,
-        vm_type: u64,
-        #[cfg(feature = "sev_snp")] _mem_size: u64,
-    ) -> hypervisor::Result<Arc<dyn vm::Vm>> {
-        self.create_vm_with_type(vm_type)
-    }
-
     /// Create a KVM vm object of a specific VM type and return the object as Vm trait object
     ///
     /// # Examples
     ///
     /// ```
     /// # use hypervisor::kvm::KvmHypervisor;
-    /// use hypervisor::kvm::KvmVm;
+    /// # use hypervisor::kvm::KvmVm;
+    /// # use hypervisor::HypervisorVmConfig;
     /// let hypervisor = KvmHypervisor::new().unwrap();
-    /// let vm = hypervisor.create_vm_with_type(0).unwrap();
+    /// let vm = hypervisor.create_vm(HypervisorVmConfig::default()).unwrap();
     /// ```
-    fn create_vm_with_type(&self, vm_type: u64) -> hypervisor::Result<Arc<dyn vm::Vm>> {
+    fn create_vm(&self, _config: HypervisorVmConfig) -> hypervisor::Result<Arc<dyn vm::Vm>> {
```

Comment:
nit: `_config` -> `config`. There are quite few similar changes that need to be updated.

Reply:
If we don't enable tdx or sev_snp this complains  unused variable.

Reply:
Aha. You are right.